tools/README.md

```diff
@@ -0,0 +1,135 @@
+# Introduction
+
+This directory contains some documentation on how HTML and PDF
+versions of the P4.org specifications are generated and published on
+the P4.org web site.
+
+
+## Released versions of specifications
+
+Since new released versions of P4.org specifications are typically
+released about once per year, or less often, all of the released
+versions are generated, and copied to separate files on the P4.org web
+server.
+
+This includes the latest released versions in the first section of
+this page that has links to specifications: https://p4.org/specs/ It
+also includes all of the links in the "Archives" section at the end of
+that page.
+
+
+## Working draft versions of specifications
+
+We only publish the most recent HTML and PDF draft version of
+specifications, generated from the latest commit in the appropriate
+specification repository.  If someone wants an older version, they can
+check out any older version of the specification soruces they wish,
+and then follow the published instructions for how to generate HTML
+and PDF from the source,
+e.g. [here](https://github.com/p4lang/p4-spec/tree/main/p4-16/spec#building).
+
+
+### Working draft versions of specifications, generated from AsciiDoc source
+
+The details of this are still to be decided.  Ideally they should be
+documnted here after those decisions have been made.
+
+
+### Working draft versions of specifications, generated from Madoko source (historical interest only as of 2024)
```

Comment:
The links are still relevant today, as the Asciidoc CI mechanisms are still publishing files to these URLs.